Fedora 39 Build Issue

Documenting this here in hope that someone (including the rubber ducks) can help me wrap my brain around what has happened here.

Commit at 896c47a982aa0a1ff057e61e0d233c1573b25ab8 was building just fine on Fedora 36 (glibc 2.35 + Swift 5.8.0). When the same commit is built on Fedora 39 (glibc 2.38 + Swift 5.8.0) it blows up with complication errors:

```
/home/grant/Projects/priv/swiftarr/.build/checkouts/swift-log/Sources/Logging/Logging.swift:1392:64: error: value of optional type 'UnsafeMutablePointer<tm>?' must be unwrapped to a value of type 'UnsafeMutablePointer<tm>'
        strftime(&buffer, buffer.count, "%Y-%m-%dT%H:%M:%S%z", localTime)
```

Description and "workaround" given in https://github.com/apple/swift-log/issues/278 (https://github.com/JCWasmx86/swift-log/commit/332ed31d777c7acbf4790a790bcfb54fe264de27)
We then hit:

```
/home/grant/Projects/priv/swiftarr/.build/checkouts/swift-nio-ssl/Sources/NIOSSL/PosixPort.swift:42:64: error: cannot convert value of type '(UnsafeMutablePointer<FILE>) -> Int32' (aka '(UnsafeMutablePointer<_IO_FILE>) -> Int32') to specified type '@convention(c) (FILEPointer?) -> CInt' (aka '@convention(c) (Optional<UnsafeMutablePointer<_IO_FILE>>) -> Int32')
private let sysFclose: @convention(c) (FILEPointer?) -> CInt = fclose
```

Description and "workaround" given in https://github.com/apple/swift-nio-ssl/issues/429
Finally we hit our own code issues:

```
/home/grant/Projects/priv/swiftarr/Sources/swiftarr/Image/Image.swift:359:18: error: value of optional type 'UnsafeMutablePointer<FILE>?' (aka 'Optional<UnsafeMutablePointer<_IO_FILE>>') must be unwrapped to a value of type 'UnsafeMutablePointer<FILE>' (aka 'UnsafeMutablePointer<_IO_FILE>')
...
/home/grant/Projects/priv/swiftarr/Sources/swiftarr/Image/Image.swift:393:18: error: value of optional type 'UnsafeMutablePointer<FILE>?' (aka 'Optional<UnsafeMutablePointer<_IO_FILE>>') must be unwrapped to a value of type 'UnsafeMutablePointer<FILE>' (aka 'UnsafeMutablePointer<_IO_FILE>')
```

Both of which can be fixed by force-unwrapping the various file handles in the respective calls.

It seems that all of this stems from some kind of change in the way Swift interacts with newer (>2.35) glibc.

--- a/Package.resolved
+++ b/Package.resolved
@@ -221,8 +221,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/apple/swift-log.git",
       "state" : {
-        "revision" : "532d8b529501fb73a2455b179e0bbb6d49b652ed",
-        "version" : "1.5.3"
+        "revision" : "e97a6fcb1ab07462881ac165fdbb37f067e205d5",
+        "version" : "1.5.4"
       }
     },
     {
@@ -266,8 +266,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/apple/swift-nio-ssl.git",
       "state" : {
-        "revision" : "320bd978cceb8e88c125dcbb774943a92f6286e9",
-        "version" : "2.25.0"
+        "revision" : "7c381eb6083542b124a6c18fae742f55001dc2b5",
+        "version" : "2.26.0"
       }
     },
     {

